rubix-tests/src/test/java/com/qubole/rubix/tests/TestThriftServerJVM.java

```diff
@@ -179,4 +184,49 @@ public void testCreateBookKeeperClient_OnNotRunningPort() throws IOException, In
     String host = "localhost";
     bookKeeperFactory.createBookKeeperClient(host, conf);
   }
+
+  public void testBookKeeperHealth() throws IOException, InterruptedException, TTransportException, TException
+  {
+    String host = "localhost";
+
+    RetryingBookkeeperClient client;
+    client = bookKeeperFactory.createBookKeeperClient(host, conf);
+
+    boolean isBookKeeperAlive = client.isBookKeeperAlive();
+    assertTrue(isBookKeeperAlive == true, "BookKeeper is not Alive");
+  }
+
+  @Test(enabled = true, expectedExceptions = org.apache.thrift.shaded.transport.TTransportException.class)
```

Comment:
This test is not any justice here. We should check the connectivity of the daemon running on the same port. By running on a different port is not a valid use case here.

Reply:
Even though the fix is about connectivity of running daemon, this test case tests if process is exiting with false if the daemon is not present at that port. So, I think it is okay to have it here because it adds as one of the valid test-case even though not related to fix.